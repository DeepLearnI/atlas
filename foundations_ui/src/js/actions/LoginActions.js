import React from 'react';
import { Redirect } from 'react-router-dom';
import BaseActions from './BaseActions';

class LoginActions {
  static getLogin(username, password) {
    const encodedString = btoa(`${username}:${password}`);

    const url = 'auth/cli_login';
    return BaseActions.getFromStagingAuth(url, encodedString);
  }

  static postLogin(body) {
    const url = 'login';
    return BaseActions.postToAPI(url, body);
  }

  static redirect(urlName) {
    return <Redirect push to={urlName} />;
  }
}
export default LoginActions;
