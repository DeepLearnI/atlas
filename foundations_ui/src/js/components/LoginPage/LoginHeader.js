import React, { Component } from 'react';
import PropTypes from 'prop-types';

class LoginHeader extends Component {
  constructor(props) {
    super(props);
    this.state = {
      pageTitle: this.props.pageTitle,
    };
  }

  componentWillReceiveProps(nextProps) {
    this.setState({ pageTitle: nextProps.pageTitle });
  }

  render() {
    const { pageTitle } = this.state;

    return (
      <div className="project-header-container">
        <div className="project-header-logo-container">
          <div className="i--icon-logo" />
          <h2 className="font-bold">Foundations</h2>
        </div>
        <div className="project-header-info-container">
          <div className="project-header-total-projects-container">
            <div className="half-width inline-block">
              <h1 className="blue-border-bottom font-bold">{ pageTitle }</h1>
            </div>
          </div>
          <div className="project-header-sort-filter-container" />
        </div>
      </div>
    );
  }
}

LoginHeader.propTypes = {
  pageTitle: PropTypes.string,
};

LoginHeader.defaultProps = {
  pageTitle: '',
};

export default LoginHeader;
