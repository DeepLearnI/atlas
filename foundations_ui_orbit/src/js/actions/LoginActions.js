import React from 'react';
import { Redirect } from 'react-router-dom';
import { getFromStagingAuth } from './BaseActions';

class LoginActions {
  static getLogin(username, password) {
    const encodedString = btoa(`${username}:${password}`);

    const url = 'auth/cli_login';
    return getFromStagingAuth(url, encodedString);
  }

  static redirect(urlName) {
    return <Redirect push to={urlName} />;
  }
}
export default LoginActions;
