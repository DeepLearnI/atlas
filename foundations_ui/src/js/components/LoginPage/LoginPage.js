import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';
import Cookies from 'js-cookie';
import LoginActions from '../../actions/LoginActions';
import ErrorMessage from '../common/ErrorMessage';
import CommonHeader from '../common/CommonHeader';


class LoginPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoggedIn: null,
      loginResponse: [],
      username: '',
      password: '',
    };
    this.handleUsernameChange = this.handleUsernameChange.bind(this);
    this.handlePasswordChange = this.handlePasswordChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.login = this.login.bind(this);
  }

  handleUsernameChange(event) {
    this.setState({
      username: event.target.value,
    });
    event.preventDefault();
  }

  handlePasswordChange(event) {
    this.setState({
      password: event.target.value,
    });
    event.preventDefault();
  }

  handleSubmit(event) {
    const { username, password, loginResponse } = this.state;
    this.login(username, password);
    event.preventDefault();
  }

  async login(username, password) {
    LoginActions.getLogin(username, password).then(res => {
      if (res.status === 200) {
        this.setState({
          loginResponse: res,
          isLoggedIn: true,
        });
      } else {
        this.setState({
          loginResponse: res,
          isLoggedIn: false,
        });
      }
    });
  }

  render() {
    const {
      isLoggedIn,
      loginResponse,
      username,
      password,
    } = this.state;

    let passwordError;

    if (isLoggedIn) {
      this.props.history.push('/projects');
    }

    if (loginResponse.status === 401) {
      passwordError = 'Incorrect credentials';
    }

    if (loginResponse.status === 400) {
      return <ErrorMessage errorCode={loginResponse[0]} />;
    }

    return (
      <div className="login-page-container">
        <CommonHeader {...this.props} />
        <div className="login-body-container">
          <h3>Welcome back!</h3>
          <form onSubmit={this.handleSubmit}>
            <ul>
              <li>
                <input
                  className="login-form-username"
                  type="username"
                  name="username"
                  value={username}
                  onChange={this.handleUsernameChange}
                  placeholder="username"
                />
              </li>
              <li>
                <input
                  className="login-form-password"
                  type="password"
                  name="password"
                  value={password}
                  onChange={this.handlePasswordChange}
                  placeholder="password"
                />
              </li>
              <li>
                <input className="login-submit" type="submit" value="Login" />
              </li>
              <li>
                <p className="auth-error">{passwordError}</p>
              </li>
            </ul>
          </form>
          <h4>Don&#39;t have an account? <a href="/support">Get Started</a></h4>
        </div>
      </div>
    );
  }
}

LoginPage.propTypes = {
  isMount: PropTypes.bool,
  isLoggedIn: PropTypes.bool,
  isLoaded: PropTypes.bool,
  loginResponse: PropTypes.array,
  history: PropTypes.object,
};

LoginPage.defaultProps = {
  isMount: false,
  isLoaded: false,
  isLoggedIn: false,
  loginResponse: [],
  history: {},
};

export default LoginPage;
