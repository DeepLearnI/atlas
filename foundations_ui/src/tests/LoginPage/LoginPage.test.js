import React from 'react';
import ReactDOM from 'react-dom';
import fake from 'faker';
import LoginPage from '../../js/components/LoginPage/LoginPage';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import configureTests from '../setupTests';

import LoginActions from '../../js/actions/LoginActions';

configureTests();

it('Shallow Renders ProjectPage', () => {
  const wrapper = shallow(<LoginPage/>);
  expect(wrapper).toMatchSnapshot();
});

it('isLoggedIn is null by Default', () => {
    const wrapper = shallow(<LoginPage/>);
    expect(wrapper.state('isLoggedIn')).toEqual(null)
});


it("postLogin is called on Login", async () => {
    LoginActions.postLogin = jest.fn().mockResolvedValue([200, 'OK']);
    const wrapper = shallow(<LoginPage/>);
    const fake_data = {password: fake.lorem.word()};
    await wrapper.instance().login(fake_data)
    expect(LoginActions.postLogin).toBeCalledWith(fake_data)

});

it("Sets isLoggedIn to True on Login", async () => {
    LoginActions.postLogin = jest.fn().mockResolvedValue([200, 'OK']);
    const wrapper = shallow(<LoginPage/>);
    await wrapper.instance().login('data')
    expect(wrapper.state('isLoggedIn')).toEqual(true)

});

it("Sets isLoggedIn to False on Login", async () => {
    LoginActions.postLogin = jest.fn().mockResolvedValue([401, 'Unauthorized']);
    const wrapper = shallow(<LoginPage/>);
    const fake_data = {password: fake.lorem.word()};
    await wrapper.instance().login(fake_data)
    expect(wrapper.state('isLoggedIn')).toEqual(false)
});

it("Sets loginResponse on Login", async () => {
    LoginActions.postLogin = jest.fn().mockResolvedValue([200, 'OK']);
    const wrapper = shallow(<LoginPage/>);
    await wrapper.instance().login('data')
    expect(wrapper.state('loginResponse')).toEqual([200, 'OK'])
});

it("handleSubmit should call login function", async () => {
    const wrapper = shallow(<LoginPage/>);
    const wrapperInstance = wrapper.instance();
    const mockEvent = {
        preventDefault: () => '',
    }
    wrapperInstance.login = jest.fn().mockResolvedValue('cat')
    wrapperInstance.handleSubmit(mockEvent)
    expect(wrapperInstance.login).toBeCalledWith(new FormData())
});

it("calls redirect if isLoggedIn", () => {
    const wrapper = shallow(<LoginPage/>);
    LoginActions.redirect = jest.fn();
    wrapper.setState({
        isLoggedIn: true
    })
    expect(LoginActions.redirect).toBeCalledWith('/projects');
})