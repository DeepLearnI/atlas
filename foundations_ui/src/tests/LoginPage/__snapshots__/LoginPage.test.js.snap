// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Renders ProjectPage 1`] = `
<div
  className="login-page-container"
>
  <div
    className="header"
  >
    <Toolbar />
    <ProjectHeader
      numProjects={0}
      pageTitle="Login"
    />
  </div>
  <div
    className="login-body-container"
  >
    <form
      onSubmit={[Function]}
    >
      <label>
        Password:
        <input
          name="password"
          onChange={[Function]}
          type="password"
          value=""
        />
      </label>
      <input
        type="submit"
        value="Submit"
      />
    </form>
    <p
      className="auth-error"
    />
  </div>
</div>
`;
